main.py

```diff
@@ -155,60 +164,58 @@ def get_interfaces(self):
 
         return jsonify({'interfaces': interfaces})
 
-    @rest('v3/interfaces/enable', methods=['POST'])
+    @rest('v3/interfaces/switch/<dpid>/enable', methods=['POST'])
     @rest('v3/interfaces/<interface_enable_id>/enable', methods=['POST'])
-    def enable_interface(self, interface_enable_id=None):
+    def enable_interface(self, interface_enable_id=None, dpid=None):
         """Administratively enable a list of interfaces in the topology."""
         error_list = []  # List of interfaces that were not activated.
         interface_ids = []
         msg_error = "Some interfaces couldn't be found and activated: "
-        if not interface_enable_id:
-            interface_ids = self._get_data(request)
-        else:
-            interface_ids.append(interface_enable_id)
-        for interface_id in interface_ids:
-            switch_id = ":".join(interface_id.split(":")[:-1])
-            interface_number = int(interface_id.split(":")[-1])
-
-            try:
-                switch = self.controller.switches[switch_id]
-            except KeyError as exc:
-                return jsonify(f"Switch not found: {exc}"), 404
+        try:
+            if dpid:
+                interface_ids = self._get_interfaces(dpid)
+            else:
+                interface_ids.append(interface_enable_id)
+            for interface_id in interface_ids:
+                switch_id = ":".join(interface_id.split(":")[:-1])
+                interface_number = int(interface_id.split(":")[-1])
+            switch = self.controller.switches[switch_id]
+        except KeyError as exc:
+            return jsonify(f"Switch not found: {exc}"), 404
 
-            try:
-                switch.interfaces[interface_number].enable()
-            except KeyError as exc:
-                error_list.append(f"Switch {switch_id} Interface {exc}")
+        try:
+            switch.interfaces[interface_number].enable()
+        except KeyError as exc:
+            error_list.append(f"Switch {switch_id} Interface {exc}")
```

Comment:
Here you need to traverse the whole list of interfaces to enable them all, or just one interface.

```suggestion
        if interface_enable_id:
            interface_number = int(interface_enable_id.split(":")[-1])
            try:
                switch.interfaces[interface_number].enable()
            except KeyError as exc:
                error_list.append(f"Switch {switch_id} Interface {exc}")
        else:
            for interface in switch.interfaces.values():
                interface.enable()
```